lib/posix/timer.c

```diff
@@ -39,6 +46,29 @@ static void zephyr_timer_wrapper(struct k_timer *ztimer)
 	(timer->sigev_notify_function)(timer->val);
 }
 
+static void *zephyr_thread_wrapper(void *arg)
+{
+	struct timer_obj *timer = (struct timer_obj *)arg;
+
+	while (1) {
+		if (timer->reload == 0U) {
+			timer->status = NOT_ACTIVE;
+		}
+
+		pthread_cond_wait(&timer->cond, &timer->lock);
```

Comment:
use of pthread_cond_wait is wrong.. read man page

Reply:
I'm not convinced using pthread_cond with mutex, why not simple binary semaphore !